extensions/src/tables/index.ts

```diff
@@ -49,9 +49,30 @@ export default class Tables extends Extension<Details, Blocks> {
       };
     },
     onLoad: (self, data) => {
-      self.tables = data.tables;
-      self.rowNames = data.rowNames;
```

Comment:
Some beautiful code below! I'm not even 'mad' at it, and think this would be fine to include. 

If you wanted to be a little fancier, you could make use of some modern javascript functionality to make it a little more concise, which may or may not make this more clunky to maintain in the future:

```ts
self.tables = data.tables;
self.rowNames = data.rowNames ?? Object.keys(self.tables).reduce((rowNames, tableName) => {
        rowNames[tableName] = data.tables[tableName].map((row, i) => `row${i + 1}`);
        return rownNames;
}, {});
// similiar for columnNames
```

Reply:
Hmm it won't work like this exactly, because if `data.rowNames` isn't defined, then neither is `data.tables`, and so it'd need to be `const tableNames = Object.keys(data);`, since `data` is the tables data in older projects.

I think keeping the singular conditional but using your reduce code would be the happy medium - I'll make that change now!